src/main/java/org/opentripplanner/apis/transmodel/model/timetable/DatedServiceJourneyQuery.java

```diff
@@ -93,72 +94,39 @@ public static GraphQLFieldDefinition createQuery(
           .type(new GraphQLList(new GraphQLNonNull(Scalars.GraphQLString)))
       )
       .dataFetcher(environment -> {
-        Stream<TripOnServiceDate> stream = GqlUtil
-          .getTransitService(environment)
-          .getAllTripOnServiceDates()
-          .stream();
-
+        var authorities = mapIDsToDomainNullSafe(environment.getArgument("authorities"));
         var lines = mapIDsToDomainNullSafe(environment.getArgument("lines"));
         var serviceJourneys = mapIDsToDomainNullSafe(environment.getArgument("serviceJourneys"));
+        var replacementFor = mapIDsToDomainNullSafe(environment.getArgument("replacementFor"));
         var privateCodes = environment.<List<String>>getArgument("privateCodes");
         var operatingDays = environment.<List<LocalDate>>getArgument("operatingDays");
         var alterations = environment.<List<TripAlteration>>getArgument("alterations");
-        var authorities = mapIDsToDomainNullSafe(environment.getArgument("authorities"));
-        var replacementFor = mapIDsToDomainNullSafe(environment.getArgument("replacementFor"));
 
-        if (!lines.isEmpty()) {
-          stream =
-            stream.filter(tripOnServiceDate ->
-              lines.contains(tripOnServiceDate.getTrip().getRoute().getId())
-            );
+        if (CollectionUtils.isEmpty(operatingDays)) {
+          throw new IllegalArgumentException("At least one operatingDay must be provided.");
         }
 
-        if (!serviceJourneys.isEmpty()) {
-          stream =
-            stream.filter(tripOnServiceDate ->
-              serviceJourneys.contains(tripOnServiceDate.getTrip().getId())
-            );
+        TripOnServiceDateRequestBuilder tripOnServiceDateRequestBuilder = TripOnServiceDateRequest
+          .of()
+          .withOperatingDays(operatingDays)
+          .withAuthorities(authorities)
+          .withLines(lines)
+          .withServiceJourneys(serviceJourneys)
+          .withReplacementFor(replacementFor);
+
+        if (!CollectionUtils.isEmpty(privateCodes)) {
+          tripOnServiceDateRequestBuilder =
+            tripOnServiceDateRequestBuilder.withPrivateCodes(privateCodes);
         }
```

Comment:
With a null-safe builder we could simplify:
```suggestion
        tripOnServiceDateRequestBuilder =
          tripOnServiceDateRequestBuilder.withPrivateCodes(privateCodes);
```

Reply:
Implementing.